classes/class3/README.md

```diff
@@ -1 +1,364 @@
+* Handling templates. We haven't gotten here yet, but we'll touch on this later.
+Let's go back to our `app.js`. We want more than just hello world. Let's make Express show the string `hello olin` when we go to `http://localhost:3000/olin`.
+
+Below the `app.get('/',...` statement, add the following:
+```javascript
+app.get('/olin', function(req, res){
+  res.send('hello olin');
+});
+```
+
+So what does `app.get` do? It tells express that every time we have a `GET` request from a client (the browser). We'll learn more about this in Routing.
+###Routing
+Routing is the process of serving up different pages for different urls. When you go to www.mycoolsite.com/ your computer goes out on the internet and asks mycoolsite's server for a page. Mycoolsite's server then sees that request and sends back information to your computer in the form of html. This html is then rendered on your browser.
+
+If you go to www.mycoolsite.com/ mycoolsite's servers obviously can't send you the same data it sent www.mycoolsite.com/olin. So mycoolsite's servers needs to differentiate `/` from `/olin`. This process is known as routing.
+
+In the Node Beginner Book, we did routing through something like
+```javascript
+var pathname = url.parse(request.url).pathname;
+route(handle, pathname, response, request);
+```
+Instead of writing the code to handle the route ourselves, Express does it for us with these statements:
+```javascript
+app.get('/', function (req, res){
+  res.send('hello world');
+});
+
+app.get('/olin', function (req, res){
+  res.send('hello olin');
+});
+```
+These two routes we created are for the index page (/) and the olin page (/olin).
+
+So what does `app.get` do? It tells express that every time that particular route (the first string argument) receives a `HTTP GET` request, we want to execute the anonymous function (the second argument). `HTTP` lets you perform different types of requests for a particular route, and these types are called **methods**. `GET` is just one of the methods you can perform, and is the most common (every time request an image or a script for example). You can [view the other types of HTTP methods on wikipedia](http://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods). But for now, let's only consider these two:
+* `GET` returns a resource (such as an image or an html page). This is used for when your browser wants to read information from a server. Parameters and options are communicated in the url through the path (`/olin`) and query string parameters (`?pageid=12&button=2&coolsetting=super`) which you may recognize from various websites.
+* `POST` is used for when your browser wants to send information over to the server. For example, when you fill out an online form, that data is sent over to the server as `POST` data. This information is sent in the request header, is not visible in plain-text and can be encrypted.
```

Comment:
`POST` information is sent in the body and is readable in plain text. `GET` information is sent in the header as part of the path.

A brief side note on REST semantics:
`POST` is for generally effecting some change on the server (like sending login information to start a session, or executing some background process).

 `PUT` is for putting a resource at a specific URL. If you `PUT` something to `/olin/students/evan`, `GET`ing `/olin/students/evan` should return what you `PUT`.

What you wrote isn't wrong, I just wanted to say this bit so we're all on the same page. I think people generally aren't taught appropriate uses of `POST` vs. `PUT`.